### PR TITLE
fix issue: tag not creating in time

### PR DESCRIPTION
**Pull request by piggona**

### CODE REVIEW

**Comment by piggona** [comment]
✗ make testacc TEST=./tencentcloud TESTARGS=" -run=TestAccTencentCloudAsScalingGroup_full"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./tencentcloud -v -run=TestAccTencentCloudAsScalingGroup_full -timeout 120m
=== RUN   TestAccTencentCloudAsScalingGroup_full
--- PASS: TestAccTencentCloudAsScalingGroup_full (70.95s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     71.025s

**Comment by piggona** [comment]
make testacc TEST=./tencentcloud TESTARGS=" -run=TestAccTencentCloudCbsStorage_full" 
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./tencentcloud -v -run=TestAccTencentCloudCbsStorage_full -timeout 120m
=== RUN   TestAccTencentCloudCbsStorage_full
=== PAUSE TestAccTencentCloudCbsStorage_full
=== CONT  TestAccTencentCloudCbsStorage_full
--- PASS: TestAccTencentCloudCbsStorage_full (6.02s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     6.085s

**Comment by piggona** [comment]
✗ make testacc TEST=./tencentcloud TESTARGS=" -run=TestAccTencentCloudCfsFileSystem"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./tencentcloud -v -run=TestAccTencentCloudCfsFileSystem -timeout 120m
=== RUN   TestAccTencentCloudCfsFileSystemsDataSource
--- PASS: TestAccTencentCloudCfsFileSystemsDataSource (58.23s)
=== RUN   TestAccTencentCloudCfsFileSystem
--- PASS: TestAccTencentCloudCfsFileSystem (38.91s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     97.221s

**Comment by piggona** [comment]
make testacc TEST=./tencentcloud TESTARGS=" -run=TestAccTencentCloudClbInstance_open"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./tencentcloud -v -run=TestAccTencentCloudClbInstance_open -timeout 120m
=== RUN   TestAccTencentCloudClbInstance_open
=== PAUSE TestAccTencentCloudClbInstance_open
=== CONT  TestAccTencentCloudClbInstance_open
--- PASS: TestAccTencentCloudClbInstance_open (24.25s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     24.321s

**Comment by piggona** [comment]
make testacc TEST=./tencentcloud TESTARGS=" -run=TestAccTencentCloudCcnV3Basic"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./tencentcloud -v -run=TestAccTencentCloudCcnV3Basic -timeout 120m
=== RUN   TestAccTencentCloudCcnV3Basic
--- PASS: TestAccTencentCloudCcnV3Basic (17.02s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     17.085s

**Comment by piggona** [comment]
✗ make testacc TEST=./tencentcloud TESTARGS=" -run=TestAccTencentCloudRedisInstance"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./tencentcloud -v -run=TestAccTencentCloudRedisInstance -timeout 120m
=== RUN   TestAccTencentCloudRedisInstancesDataSource
--- PASS: TestAccTencentCloudRedisInstancesDataSource (57.78s)
=== RUN   TestAccTencentCloudRedisInstance
--- PASS: TestAccTencentCloudRedisInstance (165.23s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     223.084s

File: tencentcloud/resource_tc_as_scaling_group.go [hunk 1]
```diff
@@ -431,11 +431,17 @@ func resourceTencentCloudAsScalingGroupRead(d *schema.ResourceData, meta interfa
...
+	tcClient := meta.(*TencentCloudClient).apiV3Conn
+	tagService := &TagService{client: tcClient}
+	tags, err := tagService.DescribeResourceTags(ctx, "as", "auto-scaling-group", tcClient.Region, d.Id())
+	if err != nil {
+		return err
+	}
+
+	if err := d.Set("tags", tags); err != nil {
+		log.Printf("[CRITAL]%s provider set tags fail, reason:%s\n ", logId, err.Error())
+		return err
```
Comment (Sherlock-Holo) [inline tencentcloud/resource_tc_as_scaling_group.go:443]: 这个可以 _ = d.Set(tags) 忽略掉 error，一般不会出错

File: tencentcloud/resource_tc_cbs_storage.go [hunk 2]
```diff
@@ -289,6 +284,18 @@ func resourceTencentCloudCbsStorageRead(d *schema.ResourceData, meta interface{}
...
+	tagService := &TagService{client: tcClient}
+	tags, err := tagService.DescribeResourceTags(ctx, "cvm", "volume", tcClient.Region, d.Id())
+	if err != nil {
+		return err
+	}
+
+	if err := d.Set("tags", tags); err != nil {
+		log.Printf("[CRITAL]%s provider set tags fail, reason:%s\n ", logId, err.Error())
+		return err
+	}
```
Comment (Sherlock-Holo) [inline tencentcloud/resource_tc_cbs_storage.go:297]: 同上

File: tencentcloud/resource_tc_ccn.go [hunk 3]
```diff
@@ -137,6 +153,17 @@ func resourceTencentCloudCcnRead(d *schema.ResourceData, meta interface{}) error
...
+	tcClient := meta.(*TencentCloudClient).apiV3Conn
+	tagService := &TagService{client: tcClient}
+	tags, err := tagService.DescribeResourceTags(ctx, "vpc", "ccn", tcClient.Region, d.Id())
+	if err != nil {
+		return err
+	}
+
+	if err := d.Set("tags", tags); err != nil {
+		log.Printf("[CRITAL]%s provider set tags fail, reason:%s\n ", logId, err.Error())
+		return err
```
Comment (Sherlock-Holo) [inline tencentcloud/resource_tc_ccn.go:165]: 同上

File: tencentcloud/resource_tc_cfs_file_system.go [hunk 4]
```diff
@@ -220,6 +233,18 @@ func resourceTencentCloudCfsFileSystemRead(d *schema.ResourceData, meta interfac
...
+	tcClient := meta.(*TencentCloudClient).apiV3Conn
+	tagService := &TagService{client: tcClient}
+	tags, err := tagService.DescribeResourceTags(ctx, "cfs", "filesystem", tcClient.Region, d.Id())
+	if err != nil {
+		return err
+	}
+
+	if err := d.Set("tags", tags); err != nil {
+		log.Printf("[CRITAL]%s provider set tags fail, reason:%s\n ", logId, err.Error())
+		return err
```
Comment (Sherlock-Holo) [inline tencentcloud/resource_tc_cfs_file_system.go:245]: 同上

File: tencentcloud/resource_tc_clb_instance.go [hunk 5]
```diff
@@ -336,7 +335,18 @@ func resourceTencentCloudClbInstanceRead(d *schema.ResourceData, meta interface{
...
+	tcClient := meta.(*TencentCloudClient).apiV3Conn
+	tagService := &TagService{client: tcClient}
+	tags, err := tagService.DescribeResourceTags(ctx, "clb", "loadbalancerid", tcClient.Region, d.Id())
+	if err != nil {
+		return err
+	}
+
+	if err := d.Set("tags", tags); err != nil {
+		log.Printf("[CRITAL]%s provider set tags fail, reason:%s\n ", logId, err.Error())
+		return err
```
Comment (Sherlock-Holo) [inline tencentcloud/resource_tc_clb_instance.go:348]: 同上

**Comment by piggona** [comment]
1. added: schema ‘tag' as TypeMap, involving resources: CCN, CFS
2. fixed: tag not creating/updating in time, involving resources: AS, CBS, CLB, Redis

File: tencentcloud/resource_tc_as_scaling_group.go (hunk 1 above)
Comment (piggona) [reply tencentcloud/resource_tc_as_scaling_group.go]: fixed

File: tencentcloud/resource_tc_ccn.go [hunk 6]
```diff
@@ -173,6 +196,21 @@ func resourceTencentCloudCcnUpdate(d *schema.ResourceData, meta interface{}) err
...
 		}
 	}
+	if d.HasChange("tags") {
+
+		oldValue, newValue := d.GetChange("tags")
+		replaceTags, deleteTags := diffTags(oldValue.(map[string]interface{}), newValue.(map[string]interface{}))
+
+		tcClient := meta.(*TencentCloudClient).apiV3Conn
+		tagService := &TagService{client: tcClient}
+		region := meta.(*TencentCloudClient).apiV3Conn.Region
```
Comment (oliverpei) [inline tencentcloud/resource_tc_ccn.go:206]: region := tcClient.Region

File: tencentcloud/resource_tc_ccn.go [hunk 7]
```diff
@@ -173,6 +196,21 @@ func resourceTencentCloudCcnUpdate(d *schema.ResourceData, meta interface{}) err
...
+		resourceName := BuildTagResourceName("vpc", "ccn", region, d.Id())
+		err := tagService.ModifyTags(ctx, resourceName, replaceTags, deleteTags)
+		if err != nil {
+			return err
+		}
+		d.SetPartial("tags")
```
Comment (oliverpei) [inline tencentcloud/resource_tc_ccn.go:212]: 这个需要先 d.Partial(true)

File: tencentcloud/resource_tc_cfs_file_system_test.go [hunk 8]
```diff
@@ -119,5 +172,10 @@ resource "tencentcloud_cfs_file_system" "foo" {
   protocol = "NFS"
   vpc_id = tencentcloud_vpc.vpc.id
   subnet_id = tencentcloud_subnet.subnet.id
+  tags = {
```
Comment (oliverpei) [inline tencentcloud/resource_tc_cfs_file_system_test.go:175]: 了解下TF规范，块之间加个空行

File: tencentcloud/resource_tc_ccn.go (hunk 7 above)
Comment (piggona) [reply tencentcloud/resource_tc_ccn.go:212]: fixed

File: tencentcloud/resource_tc_cfs_file_system_test.go (hunk 8 above)
Comment (piggona) [reply tencentcloud/resource_tc_cfs_file_system_test.go:175]: fixed

File: tencentcloud/resource_tc_ccn.go (hunk 6 above)
Comment (piggona) [reply tencentcloud/resource_tc_ccn.go]: fixed

**Pull request merged**